caffe/include/caffe/layers/molgrid_data_layer.hpp

```diff
@@ -99,6 +99,9 @@ class MolGridDataLayer : public BaseDataLayer<Dtype> {
 
     virtual void setReceptor(const vector<atom>& receptor, const vec& translate =
         {}, const qt& rotate = {});
+
+    virtual void setReceptorFlex(const vector<atom>& receptor, const vector<vec>& flexcoords);
```

Comment:
This function can be merged with the original `setReceptor`, but the function call `setReceptor(receptor, {}, {}, flexcoords)` is a bit odd. Other signatures will change the API.

Reply:
I think the right way to do this is to create a single merged vector that's a class member of either `CNNScorer` or `model` that tracks the full set of receptor atoms (alternatively you could pass in references to the existing vectors with <start> and <end> indices, but that API isn't as clean so we'll go with the merged vector). You can have another class member that's the vector of ligand atoms so you don't have to keep calling those getter functions to make copies every time. There should be a single setReceptor function, which gets passed the correct receptor atoms vector.

Reply:
Yes, this is what I was thinking but didn't want to proceed this way before discussing. I think the best option would be to have a similar signature for `setLigand` and `setReceptor`, where both reference to an receptor atom and coordinates atom vector are passed. I'll properly split ligand and receptor atoms/coordinates on different vectors as members of `CNNScorer` (as it's already the case for the ligand).

Reply:
Ligand and receptor atoms are now split into different structures, private members of `CNNScorer`. The `setReceptor` now has the same API of the `setLigand` function, taking coordinates explicitly.